filter.h

```diff
@@ -0,0 +1,72 @@
+// Filter Module
+
+
+// ****CHECKPOINT GOALS****
+// USER INTERFACE
+// PSEUDO CODE
+//IMPLEMENTATION
+//WELL DOCUMENTATED HEADER FILE
+// DOXYGEN FILE 
+// TEMPLATE FOR C FILE
+
+
+
+/*
+@param filter_t * - a pointer of struct type filter_t
+@param offset -  describes postion of the data going into an array
+@param scale -  the parameter value that multiples with the normalized input
+
+*/
+
+
+void Filter_Init(filter_t *, uint16_t  offset, float scale); 
+
+
+
+
+/* 
+@param raw_value - quantized values stored in the array 
```

Comment:
the user doesn't need to know anything about the array internal to the module. and filter_ptr is a pointer to the filter_t structure that represents the instance of the filter module.